nemo/export/trt_llm/converter/utils.py

```diff
@@ -414,3 +414,124 @@ def split(v, tp_size, idx, dim=0):
         return np.ascontiguousarray(np.split(v, tp_size)[idx])
     else:
         return np.ascontiguousarray(np.split(v, tp_size, axis=dim)[idx])
+
+
+# Similar to split_save_weight but done on GPU for performance
+@torch.no_grad()
+def split_save_weight_gpu(key, val, config):
```

Comment:
There are also too many duplicate code here. Can we reuse some of the existing functions and add parameters and code as needed?